metaflow/metadata/metadata.py

```diff
@@ -604,4 +638,4 @@ def __init__(self, environment, flow, event_logger, monitor):
         self._monitor = monitor
         self._environment = environment
         self._runtime = os.environ.get("METAFLOW_RUNTIME_NAME", "dev")
-        self.add_sticky_tags(sys_tags=self._tags())
+        self.add_sticky_tags(sys_tags=self._get_system_info_as_tags())
```

Comment:
Won't this still add tags to the task and lower levels? Is that the intent? Is the goal to still support the older metadata services with newer versions? We could maybe check the service version and only push sticky tags below runs if older one. It would not cover case of newer service going down to older service but we said that wouldn't be a supported path.

Reply:
Re: [service downgrade support.](https://www.notion.so/outerbounds/RFC-Implementing-tag-mutation-98ef6ba39aee4aeeb9c6150b4cffea2a#29aa525b0a664420a6d855db91c09ab2).

The option I was considering was to just keep writing task tags here until a longer time in the future, when vast majority of services in the wild have upgraded (e.g. 1 year later). This will offer a soft landing for users in the service downgrade scenario in the meantime and it doesn't seem like a big lift to do it this way (in fact, less work).  There is also UI backend service which is deployed separately and has a separate direct DB read path.  Same tradeoff there.

Open to doing version check too if folks feel strongly - I think if we want to take a harder line about one-way service upgrade right now.  If we go that way, we would probably do it a bit closer to where we construct the payload to send to the metadata provider.